python/upload_with_authentication.py

```diff
@@ -8,7 +8,11 @@
 import exifread
 import time
 
-from upload import create_dirs, UploadThread, upload_file
```

Comment:
I would do this as a `try` instead. os.path does not check the PYTHONPATH as far as I know.